src/input/controller.rs

```diff
@@ -0,0 +1,150 @@
+use super::{ControllerPort, Pad, WPad};
+use super::wpad::WPadButton;
+use super::pad::PadButton;
+
+pub enum ControllerType {
+    Gamecube,
```

Comment:
Should this be GameCube or is this deliberate?

Reply:
I think this a matter of preference in this case.

Reply:
Gotcha.